src/handlers.rs

```diff
@@ -172,3 +172,72 @@ where
     }
     Ok(())
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    macro_rules! send_to {
```

Comment:
Just for my own curiosity, why is this a macro instead of a function that returns a closure?

Reply:
I found the macro easier to do the first time, but the fn makes sense too. Happy to do whatever here, the diff is real small.

```rust
    fn send_to(v: &mut Vec<u8>) -> impl FnMut(&[u8]) -> Result<()> + '_ {
        move |bs| {
            v.extend_from_slice(bs);
            Ok(())    
        }
    }
```

Reply:
Meh, either way.